pointmatcher_ros/src/StampedPointCloud.cpp

```diff
@@ -158,6 +158,21 @@ void StampedPointCloud::filterByDistance(const float distanceThreshold, const bo
   newIdToOldId.conservativeResize(Eigen::NoChange, newId);
 }
 
+void StampedPointCloud::filterByBoundingBox(const float minX, const float maxX, const float minY, const float maxY, 
+                                            const float minZ, const float maxZ, const bool keepInside) {
+  unsigned int newId = 0;
+  for (unsigned int oldId = 0; oldId < getSize(); oldId++) {
+    const Eigen::Vector3f &point = dataPoints_.features.col(oldId);
+    if ((point.x() >= minX && point.x() <= maxX &&
+         point.y() >= minY && point.y() <= maxY &&
+         point.z() >= minZ && point.z() <= maxZ) == keepInside) {
+      dataPoints_.setColFrom(newId, dataPoints_, oldId);
+      newId++;
+    }
+  }
+  dataPoints_.conservativeResize(newId);
+}
+
```

Comment:
Knowing that libpointmatcher already implements this [here](https://github.com/ethz-asl/libpointmatcher/blob/master/pointmatcher/DataPointsFilters/BoundingBox.cpp), why are we implementing it again in this point cloud wrapper?

It could be possible to instantiate a bounding box filter as other filters shown [here](https://github.com/ethz-asl/libpointmatcher/blob/master/examples/build_map.cpp#L87). What do you think?

Reply:
Thanks, this is fixed now. I also added bool return statements to all filter*() methods for consistency.